pkg/river/internal/value/decode.go

```diff
@@ -147,6 +146,62 @@ func decode(val Value, into reflect.Value) error {
 	}
 }
 
+func decodeAny(val Value, into reflect.Value) error {
```

Comment:
This allows us to guarantee the types when decoding into an `interface{}`, whereas before it could've been anything, which is fairly difficult to reason about. 

For example, if you use `interface{}` and decode an object, it always is a `map[string]interface{}` now, where before it could've been a map or a struct from another package.